### ISSUE TITLE
Code cell prefixes

### ISSUE THREAD

**Issue by chrisjsewell** [ExecutableBookProject/MyST-NB#64]
As per https://nbsphinx.readthedocs.io/en/0.5.1/code-cells.html#Code-Cells,
it would be nice to have the option of including the cell prefixes `[1]: ` (or in a format specified in the conf.py)

@akhmerov I don't think this is possible with `jupyter-sphinx` at present?

More generally, we should probably also being looking at nbsphinx (and talking to mgeiger) and thinking about what the synergies are.

**Comment by akhmerov** [comment]
No, it's not possible with jupyter-sphinx at the moment. I am wondering also whether it is a desirable feature in the context of building a book/documentation. What use cases do you imagine?

**Comment by akhmerov** [comment]
I mean in a notebook these tell you about the sequence of executions, but in a book all executions should be sequential.

Also imagine if some inputs are hidden and some shown. Should the numbering include the hidden ones or not?

**Comment by chrisjsewell** [comment]
> I am wondering also whether it is a desirable feature in the context of building a book/documentation

Well I'm not necessarily as bothered about the numbering, its more to have a clear distinction of what are inputs and what are outputs, particularly for tutorials. 

For example, here is one I have created with IPyPublish, https://ase-notebook.readthedocs.io/en/latest/user_guide/tutorial_intro.html. I think this is nicer to read than e.g. this: https://myst-nb.readthedocs.io/en/latest/use/basic.html#code-cells-and-outputs, where all the blocks kind of merge into each other

**Comment by chrisjsewell** [comment]
Also, in the latter example, it would not be clear what is executed code and what is static code, e.g.

````markdown
```python
a = 1
```

```{jupyter-execute}
b = 2
```
```

would both look the same

**Comment by chrisjsewell** [comment]
Or here I just use `In:` and `Out:`  https://aiida-crystal17.readthedocs.io/en/develop/user_guide/calc_main.html

**Comment by chrisjsewell** [comment]
I could even envisage referencing these cells, e.g.

```rst
In :cell:in:`1` we calculated something, now in :cell:in:`5` we calculate something else.
````

where the reference would link to the prefix

**Comment by akhmerov** [comment]
Both points you brought up are relevant to address, but I'm not sure whether cell numbering does that in the best way. `In:` and `Out:` seems better, and of course everything may be configurable.

---

On a practical level, it seems to make sense to add the input/output numbers to the AST as cell attributes, and perhaps also data-attribute entries to the corresponding `div`s. That doesn't seem to impede anything, and would also fit into `jupyter-sphinx`.

**Comment by akhmerov** [comment]
> I could even envisage referencing these cells,

Ah, that's neat, although I'd be concerned about unnamed references.

**Comment by chrisjsewell** [comment]
This is the configuration that nbsphinx uses: https://nbsphinx.readthedocs.io/en/0.5.1/usage.html#nbsphinx_output_prompt

**Comment by chrisjsewell** [comment]
> Ah, that's neat, although I'd be concerned about unnamed references

Yeh more robust would be labelling cells in their metadata / a label option on the `jupter-execute` directive

**Comment by chrisjsewell** [comment]
Actually, for a more subtle approach, I quite like:

```css
div.cell div.cell_input {
  border-left-color: green;
  border-left-width: medium;
}
```

<img width="455" alt="image" src="https://user-images.githubusercontent.com/2997570/76592162-6b16c280-64ea-11ea-868f-36a025d43eb5.png">

**Comment by chrisjsewell** [comment]
I've added this now in https://jupyter-cache.readthedocs.io/en/latest/using/api.html, IMO looks nicer and is easy to read through

**Comment by choldgraf** [comment]
I am a fan as well!
